### PR TITLE
[KYUUBI #5162][DOC] Add statements about contributions co-authored with AIGC in developer guide page

### LINKED ISSUE #5162: :memo: Add statements about contributions co-authored with AIGC  in developer guide page
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What type of changes will we make to the documentation?

Refactoring

### Affects Version(s)

master

### Improving the documentation

Add statements about contributions co-authored with AIGC in developer guide page and PR template

https://www.apache.org/legal/generative-tooling.html
https://github.com/apache/spark/pull/42469
https://github.com/apache/spark-website/pull/472

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lsm1**
### _Why are the changes needed?_

close #5162 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

 No

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
@lsm1 would you like to update the PR description to include this part? :)

File: docs/contributing/code/developer.md [hunk 1]
```diff
@@ -48,3 +48,12 @@ Kyuubi uses settings.md to explain available configurations.
 
 You can run `KYUUBI_UPDATE=1 build/mvn clean test -pl kyuubi-server -am -Pflink-provided,spark-provided,hive-provided -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration`
 to append descriptions of new configurations to settings.md.
+
+## Generative tooling usage
```
Comment (yaooqinn) [inline docs/contributing/code/developer.md:52]:
```suggestion
## Generative Tooling Usage
```

File: .github/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -21,3 +21,12 @@ Please clarify why the changes are needed. For instance,
 - [ ] Add screenshots for manual tests if appropriate
 
 - [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request
+
+
+### _Was this patch authored or co-authored using generative AI tooling?_
+<!--
+If a generative AI tooling has been used in the process of authoring this patch, please include
+frase 'Generated-by: ' followed by the name of the tool and its version.
```
Comment (yaooqinn) [inline .github/PULL_REQUEST_TEMPLATE:29]: frase  -> phrase

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**